api/src/server-utils.js

```diff
@@ -103,6 +103,10 @@ module.exports = {
     }
   },
 
+  rateLimited: (req, res) => {
+    respond(req, res, 429, 'Too Many Requests');
```

Comment:
While it behaves  well enough when I tested it, this is not returning JSON like other failed logins.  I think this an easy fix though - does this achieve what we want or do we need to dive deeper?

```suggestion
    respond(req, res, 429, '{"error" : "Too Many Requests"}');
```

Reply:
The first line of the `respond` function is...

```
  if (wantsJSON(req)) {
    return writeJSON(res, code, message, details);
  }
```

This tries to respect the `accepts` header of the request so you get what you asked for.

However you have found a bug in the login code! It turns out we were never specifying the "accepts" header. I've changed that and now the error handling code works as expected. Thanks for being so diligent!